pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -451,15 +451,27 @@ func monitorFor(p *Propagator, clientPool dynamic.ClientPool, resource unversion
 				// namespaces if it's namespace scoped, so leave
 				// APIResource.Namespaced as false is all right.
 				apiResource := unversioned.APIResource{Name: resource.Resource}
-				return client.Resource(&apiResource, api.NamespaceAll).List(&options)
+				// The parameter codec used by the dynamic client cannot encode
```

Comment:
>  It should probably be enhanced to error if the current object is not the requested version

The default parameter codec doesn't know the scheme, how can it infer the version of the current object?